src/BloomExe/CollectionTab/LibraryListView.cs

```diff
@@ -92,19 +93,39 @@ enum ButtonManagementStage
 			_showHistoryMenu.Visible = _showNotesMenu.Visible = Settings.Default.ShowSendReceive;
 
 			if(Settings.Default.ShowExperimentalCommands)
-				_settingsProtectionHelper.ManageComponent(_exportToXMLForInDesignToolStripMenuItem);//we are restriting it because it opens a folder from which the user could do damage
+				_settingsProtectionHelper.ManageComponent(_exportToXMLForInDesignToolStripMenuItem);//we are restricting it because it opens a folder from which the user could do damage
 			_exportToXMLForInDesignToolStripMenuItem.Visible = Settings.Default.ShowExperimentalCommands;
 
 			SetupBookDropdownIcon();
 			_bookContextMenu.Closed += _bookContextMenu_Closed;
+			_bookContextMenu.Opening += _bookContextMenu_Opening;
+		}
+
+		// adjust the context menu item visibility based on what sort of collection we're in
+		void _bookContextMenu_Opening(object sender, CancelEventArgs e)
+		{
+			var btn = (sender as ContextMenuStrip).SourceControl as Button;
+			var btnInfo = btn.Tag as BookButtonInfo;
+			if (btnInfo.IsVernacularBook)
+				return; // leave them all on
+			if (btnInfo.IsFactoryTemplate)
+				e.Cancel = true; // don't show the menu at all
+			foreach (ToolStripItem menuItem in (sender as ContextMenuStrip).Items)
+			{
+				if (menuItem == deleteMenuItem && btnInfo.IsBLibraryBook)
+					continue; // leave this one on for BloomLibrary books
+				if (menuItem == _openFolderOnDisk)
+					continue; // leave this one on (for both BloomLibrary and BloomPack books)
+				menuItem.Visible = false;
+			}
 		}
 
 		void _bookContextMenu_Closed(object sender, ToolStripDropDownClosedEventArgs e)
 		{
-			// make these controls visible again so they are available in the right-click menu
-			toolStripSeparator1.Visible = true;
```

Comment:
I suppose we don't need to hide any separators to make it look right in any of the variations?